sb-update

```diff
@@ -45,6 +47,69 @@ def update_and_upgrade():
     subprocess.check_call(['apt', 'upgrade', '-y'], stdout=subprocess.DEVNULL)
 
 
+def validate_deb(deb_path: pathlib.Path) -> None:
+    """
+    Validate that the given path is indeed a debian package.
+
+    Raises ``ValueError`` if the package is not valid.
+    """
+    check_deb = subprocess.call(
+        ['dpkg', '--info', str(deb_path)],
+        stdout=subprocess.DEVNULL,
+    )
+
+    if check_deb != 0:
+        raise ValueError("file {} is not a deb file!".format(deb_path))
+
+
+def _extract_debs(
+    update_file: pathlib.Path,
+    working_dir: tempfile.TemporaryDirectory,
+) -> List[pathlib.Path]:
+
+    subprocess.check_call(
+        [
+            'tar',
+            '--extract',
+            '--xz',
+            '--file', str(update_file),
+        ],
+        cwd=str(working_dir),
+    )
+
+    return list(working_dir.glob("*.deb"))
+
+
+def add_debs_to_local_repo_from(update_file: pathlib.Path) -> None:
+    """
+    Given the path to an xz-compressed tarball, copy any .deb files to the
+    ``SB_DEBS`` directory.
+    """
+
+    with tempfile.TemporaryDirectory(prefix='updates-') as updates_dir:
+        found_debs = _extract_debs(update_file, pathlib.Path(updates_dir))
+
+        if not found_debs:
+            raise FileNotFoundError(
+                "Failed to find deb files in {}, exiting".format(update_file),
+            )
+
+        logging.debug("Found {} potential packages".format(len(found_debs)))
+
+        # Check that all the debs are valid
+        for deb_path in found_debs:
+            validate_deb(deb_path)
+
+        # Add all the debs to the repo directory.
+        # Note that we do this _separately_ after the checking, so that we can
+        # be somewhat confident that either all the debs that were in the
+        # update are all in the repo or none are.
+        SB_DEBS_PATH = pathlib.Path(SB_DEBS)
```

Comment:
I'd prefer this variable moved outside function scope, and with the constants at the top